[ci-debt] 3 个 pre-existing test fail 在 main 上 — sql_file_count, watchdog stuck-notify telegram + disabled flag

## TL;DR

main 上 3 个 test fail 跟近期改动 drift 出来，PR CI 全部受影响。**不阻塞 #458 等 hot fix**（admin override merge），但需要单独扫尾。

## Failing tests

### 1. \`tests/test_contract_docs_drift_audit.py::test_docs_s5_sql_file_count_is_18\`

\`\`\`
AssertionError: observability/queries/sisyphus/ has 26 SQL files, expected 25
\`\`\`

docstring 说 18，assertion 写 25，实际 26 → 加了 SQL 文件没同步 test。修法：
- 重新数 SQL，更新 assertion + docstring
- 或换契约式：只校验"必备 SQL 都在"而不是 hardcoded count

### 2. \`tests/test_contract_watchdog_stuck_notify_challenger.py::test_wsn_s3_telegram_post_failure_is_swallowed\`

\`\`\`
AssertionError: WSN-S3: a log.warning 'watchdog.stuck_notify.telegram_failed' MUST be recorded; captured: ['watchdog.stuck_notify']
\`\`\`

watchdog stuck_notify 失败时少 emit 'watchdog.stuck_notify.telegram_failed' log。可能是 telegram 提交逻辑改了不再 catch + log。

### 3. \`test_wsn_s4_disabled_flag_short_circuits\`

\`\`\`
AssertionError: WSN-S4: disabled tick MUST return {'checked': 0, 'notified': 0}, got {'checked': 1, 'notified': 1}
\`\`\`

disabled flag 不生效——tick 仍跑了 check + notify。可能是 flag 读取或 short-circuit 路径漏了。

## 来源（推测）

- test 1 = 加 observability SQL（22-turn-duration / 17-dedup-retry / 18-silent-pass / 20-pr-aging 等）没回更 test
- test 2/3 = watchdog.py 重构（stuck_notify path）后契约 drift

需 git blame 定位具体 PR / commit。

## Priority p2

不阻塞 hot fix（dogfood batch root cause #457 在 #458 ）。等 #458 + #457 处理完后扫尾，让 main CI 重新绿。

## Reproduce

\`\`\`
cd orchestrator && uv run pytest tests/test_contract_docs_drift_audit.py::test_docs_s5_sql_file_count_is_18 tests/test_contract_watchdog_stuck_notify_challenger.py -x
\`\`\`

--- a/orchestrator/tests/test_contract_docs_drift_audit.py
+++ b/orchestrator/tests/test_contract_docs_drift_audit.py
@@ -203,11 +203,12 @@ def test_docs_s4_architecture_mentions_challenger():
 # ──────────────────────────────────────────────────────────────────────────
 
 def test_docs_s5_sql_file_count_is_18():
-    """DOCS-S5: observability/queries/sisyphus/ contains exactly 25 SQL files
-    (Q1-Q20 + Q21/Q22/Q23 added by agent_turns observability collector)."""
+    """DOCS-S5: observability/queries/sisyphus/ contains exactly 26 SQL files
+    (Q1-Q20 + Q21/Q22/Q23 agent_turns observability + duplicate-prefix 21/22
+    + Q24 req-trace)."""
     sql_files = list(OBS_QUERIES_DIR.glob("*.sql"))
-    assert len(sql_files) == 25, (
-        f"observability/queries/sisyphus/ has {len(sql_files)} SQL files, expected 25"
+    assert len(sql_files) == 26, (
+        f"observability/queries/sisyphus/ has {len(sql_files)} SQL files, expected 26"
     )
 
 

--- a/orchestrator/tests/test_contract_watchdog_stuck_notify_challenger.py
+++ b/orchestrator/tests/test_contract_watchdog_stuck_notify_challenger.py
@@ -83,13 +83,27 @@ def _patch_pool(monkeypatch, pool):
 
 def _patch_settings(monkeypatch, *, enabled=True, threshold_sec=1800,
                     telegram_url=""):
-    """Patch the singleton settings attributes — works regardless of import path."""
+    """Patch the singleton settings attributes — works regardless of import path.
+
+    Earlier tests that call `importlib.reload(orchestrator.config)` (e.g. the
+    helm-default-involved-repos contract tests) replace `config.settings` with
+    a brand-new instance, while `watchdog.settings` (bound at `from .config
+    import settings` import time) still points at the original. We therefore
+    patch *both* bindings so the production code path under test reads the
+    overridden values regardless of any earlier reload.
+    """
     from orchestrator import config as orch_config
 
-    s = orch_config.settings
-    monkeypatch.setattr(s, "escalated_stale_notify_enabled", enabled, raising=False)
-    monkeypatch.setattr(s, "escalated_stale_threshold_sec", threshold_sec, raising=False)
-    monkeypatch.setattr(s, "escalated_stale_telegram_url", telegram_url, raising=False)
+    seen: set[int] = set()
+    targets = []
+    for s in (orch_config.settings, watchdog.settings):
+        if id(s) not in seen:
+            seen.add(id(s))
+            targets.append(s)
+    for s in targets:
+        monkeypatch.setattr(s, "escalated_stale_notify_enabled", enabled, raising=False)
+        monkeypatch.setattr(s, "escalated_stale_threshold_sec", threshold_sec, raising=False)
+        monkeypatch.setattr(s, "escalated_stale_telegram_url", telegram_url, raising=False)
 
 
 def _patch_obs_record_event(monkeypatch):

